db/crud.go

```diff
@@ -870,6 +870,30 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+func (db *DatabaseCollectionWithUser) computeHLVLogic(d *Document) (*Document, error) {
+	newVVEntry := CurrentVersionVector{}
+	bucketUUID, err := db.dbCtx.Bucket.UUID()
+	if err != nil {
+		return nil, err
+	}
+	newVVEntry.SourceID = bucketUUID
+	// add cas here this will be macro expanded
+	// need to keep same value for now or adding new version vector entry will fail with new cas < than old cas
+	newVVEntry.VersionCAS = d.VersionVector.Version // this will be macro expanded, need to keep same value for now
+	err = d.VersionVector.AddVersion(newVVEntry)
+	if err != nil {
+		return nil, err
+	}
+
+	persistedHLV, err := d.VersionVector.convertHLVToPersistedFormat()
```

Comment:
As mentioned elsewhere, I think we want to ensure we're only generating the persisted format at persistence time.